qiskit/result/postprocess.py

```diff
@@ -182,6 +182,12 @@ def format_statevector(vec, decimals=None):
     Returns:
         list[complex]: a list of python complex numbers.
     """
+    if type(vec).__name__ == "Statevector":
+        # Duck-type check since we can't import Statevector here
+        # without introducing cyclic dependencies.
```

Comment:
You can do this - you can protect the import inside the method and tell `pylint` to ignore the cyclical dependency.  It's done a lot elsewhere in the codebase for things like this.  Something like

```python
def format_statevector(vec, decimals=None):
    from qiskit.quantum_info import Statevector  # pylint: disable=cyclic-import
    ...
```

Actually _having_ the cyclic dependencies isn't ideal, but the ship's sailed on that one in Qiskit.

Reply:
If you think this is preferable I can change to do that

Reply:
Yeah, it's the correct way to handle this sort of thing - type checking by the name of the class is quite fragile at best.